chain/chain/src/chain.rs

```diff
@@ -3352,7 +3352,7 @@ pub fn collect_receipts_from_response(
 pub fn check_refcount_map(chain: &mut Chain) -> Result<(), Error> {
     let head = chain.head()?;
     let mut block_refcounts = HashMap::new();
-    for height in chain.store().get_genesis_height() + 1..=head.height {
+    for height in chain.store().get_genesis_height() + 1..=head.height + 100 {
```

Comment:
Where does 100 come from?

Reply:
Usually we don't create tests with more than 200-300 blocks. I guess, + 100 should cover any sanity case.

Reply:
I don't understand. Why do you need to go past head height at all?

Reply:
<img width="870" alt="Screen Shot 2020-03-31 at 7 30 37 PM" src="https://user-images.githubusercontent.com/5685493/78051281-201ffa80-7386-11ea-88dd-8a7c0f8d0bde.png">
Running tests/adversarial/start_from_genesis.py overtake doomslug_off, test0_finished/stdout:
<img width="227" alt="Screen Shot 2020-03-31 at 7 31 08 PM" src="https://user-images.githubusercontent.com/5685493/78051339-3463f780-7386-11ea-93bc-7443b3f86e8e.png">
I suppose it's related to the idea that we can receive blocks which are ahead of current head or even build a separate chain. It seems `head.height` is a bad indicator of what height is the latest block in the storage. `latest_known` doesn't work this way too.

Reply:
Could you suggest another solution for failing Nightly tests?

Reply:
Which one failed because of this?

Reply:
Some of them. Please check Nightly failures.
This one may also help: https://near-protocol.slack.com/archives/CR4NE2MCK/p1585657931003400

Reply:
@bowenwang1996 any progress here?
It's possible to apply some workarounds to ignore blocks with heights higher than `head.height`, however I not agree with this way of problem solving. Why we should not check a part of Reference Map where heights are higher than `head.height`?

Reply:
I believe this problem will cease to exist when we have the new doomslug where block with highest height is the head.

Reply:
Is there an issue related to the Doomslug update? Could you create it if not?

Reply:
https://github.com/nearprotocol/nearcore/issues/2342

Reply:
@bowenwang1996 I've updated this PR
Before #2342 we will process +100 heights in refmap to check that refmap is calculating properly. After #2342 we will remove unnecessary validations.